[WPE][GTK] when flatpak sandbox unavailable, processes are run unsandboxed with no warning

mcatanzaro opened the pull request:
#### 9fc6b8810c5244dc3acfe90ee9962b9706b637cb
<pre>
[WPE][GTK] when flatpak sandbox unavailable, processes are run unsandboxed with no warning
<a href="https://bugs.webkit.org/show_bug.cgi?id=278578">https://bugs.webkit.org/show_bug.cgi?id=278578</a>

Reviewed by Carlos Garcia Campos.

If we&apos;re running under flatpak, but flatpak-spawn doesn&apos;t work, it&apos;s
probably better to crash rather than continue to run unsandboxed. This
probably indicates a developer is experimenting, but it could be serious
if it happens by accident for an end user.

* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
(WebKit::isFlatpakSpawnUsable):
(WebKit::flatpakSpawn):
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::isFlatpakSpawnUsable): Deleted.

Canonical link: <a href="https://commits.webkit.org/283361@main">https://commits.webkit.org/283361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f17f9b371c865ee9d2460705cba488257e2eb67d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66074 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45447 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18693 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70106 "Hash f17f9b37 for PR 33206 does not build (failure)") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16684 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53246 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16965 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/70106 "Hash f17f9b37 for PR 33206 does not build (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11613 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69141 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41931 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57203 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/70106 "Hash f17f9b37 for PR 33206 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38602 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15560 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60495 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14927 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71808 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10029 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14339 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/60348 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10061 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57265 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/60639 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14573 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8284 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/1924 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41255 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42331 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43514 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42075 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
```diff
@@ -30,12 +30,35 @@
 GRefPtr<GSubprocess> flatpakSpawn(GSubprocessLauncher* launcher, const WebKit::ProcessLauncher::LaunchOptions& launchOptions, char** argv, int childProcessSocket, int pidSocket, GError** error)
 {
     ASSERT(launcher);
+    RELEASE_ASSERT(isFlatpakSpawnUsable());
```
I'm fine, but I guess we could show a warning message before crashing

mcatanzaro replied:
I'll change it to use `g_error()` before landing.

mcatanzaro committed:
[WPE][GTK] when flatpak sandbox unavailable, processes are run unsandboxed with no warning
https://bugs.webkit.org/show_bug.cgi?id=278578

Reviewed by Carlos Garcia Campos.

If we're running under flatpak, but flatpak-spawn doesn't work, it's
probably better to crash rather than continue to run unsandboxed. This
probably indicates a developer is experimenting, but it could be serious
if it happens by accident for an end user.

* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
(WebKit::isFlatpakSpawnUsable):
(WebKit::flatpakSpawn):
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::isFlatpakSpawnUsable): Deleted.

Canonical link: https://commits.webkit.org/283361@main
